lib/matplotlib/transforms.py

```diff
@@ -2984,3 +2984,54 @@ def offset_copy(trans, fig=None, x=0.0, y=0.0, units='inches'):
         y /= 72.0
     # Default units are 'inches'
     return trans + ScaledTranslation(x, y, fig.dpi_scale_trans)
+
+
+class OffsetRotation(Affine2DBase):
+    """
+    A transformation that rotates by *theta*, applying an offset
+    based on *theta_zero_location* and *theta_direction*.
+    This transformation adjusts the angle (theta) and applies a rotational
+    transformation using the transformed value.
+    """
+    def __init__(self, theta, r, trans_shift, theta_zero_location=0,
+                 theta_direction=1):
+        super().__init__()
+        self._theta = theta
+        self._r = r
+        self._trans_shift = trans_shift
+        self._theta_zero_location = theta_zero_location
+        self._theta_direction = theta_direction
+        self._mtx = None
+        self._invalid = True
+
+    def get_matrix(self):
+        if self._invalid:
+            self._mtx = np.identity(3)
+
+            # Apply the transShift transformation to adjust theta and r
+            transformed_coords = self._trans_shift.transform(
+                [[self._theta, self._r]])[0]
+
+            transformed_theta = transformed_coords[0]  # Extract transformed theta
+            transformed_r = transformed_coords[1]  # Extract transformed r
+
+            # Adjust theta based on theta_zero_location and theta_direction
+            adjusted_theta = self._theta_direction * (
+                transformed_theta - self._theta_zero_location)
```

Comment:
I don't understand this; is it not undoing what `trans_shift.transform` did?

Reply:
Hi @QuLogic, I apologize for the confusion and circular logic in my previous push. While it technically worked, I was reapplying transformations unnecessarily. I’ve since resolved this and cleaned up the approach to ensure the transformations are handled correctly, without redundant steps.